File: gr6j-core/examples/simple_model.rs
extern crate gr6j;

use chrono::NaiveDate;
use gr6j::inputs::{CatchmentData, CatchmentType, GR6JModelInputs, ModelPeriod, RunOffUnit};
use gr6j::model::GR6JModel;
use gr6j::parameter::{Parameter, X1, X2, X3, X4, X5, X6};
use gr6j::utils::example::load_data;
use log::LevelFilter;
use std::path::Path;

fn main() -> Result<(), Box<dyn std::error::Error>> {
    // Enable logging
    env_logger::builder().filter_level(LevelFilter::Info).init();

    // Collect hydrological data.
    let data = load_data()?;

    // Configure the model
    let start = NaiveDate::from_ymd_opt(1994, 1, 1).unwrap();
    let end = NaiveDate::from_ymd_opt(1998, 12, 31).unwrap();

    let inputs = GR6JModelInputs {
        time: &data.time,
        precipitation: &data.precipitation,
        evapotranspiration: &data.evapotranspiration,
        catchment: CatchmentType::OneCatchment(CatchmentData {
            area: 1.0,
            x1: X1::new(31.0)?,
            x2: X2::new(3.47)?,
            x3: X3::new(32.0)?,
            x4: X4::new(2.1)?,
            x5: X5::new(0.55)?,
            x6: X6::new(5.3)?,
            store_levels: None,
        }),
        run_period: ModelPeriod::new(start, end)?,
        warmup_period: None,
        destination: Some(Path::new(r"./gr6j-core/examples/results").to_path_buf()),
        observed_runoff: Some(&data.observed_runoff),
        run_off_unit: RunOffUnit::NoConversion,
    };
    let mut model = GR6JModel::new(inputs)?;
    model.run()?;
    Ok(())
}

File: gr6j-python/examples/simple_model.py
import logging

import pandas as pd
from datetime import date
from gr6j import (
    X1,
    X2,
    X3,
    X4,
    X5,
    X6,
    GR6JModelInputs,
    CatchmentData,
    ModelPeriod,
    RunOffUnit,
    GR6JModel,
)

# Enable module logging
logging.basicConfig(format="[%(asctime)-15s] %(levelname)s %(message)s")
logging.getLogger().setLevel(logging.INFO)

# Read the input data
data = pd.read_csv(
    r"../../gr6j-core/src/test_data/airGR_L0123001_dataset.csv",
    index_col=[0],
    parse_dates=True,
    dayfirst=True,
)

# Configure the model
start = date(1990, 1, 1)
end = date(1994, 12, 31)

# Define the catchment data
catchment = CatchmentData(
    area=1.0, x1=X1(31), x2=X2(3.47), x3=X3(32), x4=X4(2.1), x5=X5(0.55), x6=X6(5.3)
)
# you can also access the unit and range of each parameter using
print(f"Allowed range: {X1.min()}-{X1.max()} (X1.unit())")

inputs = GR6JModelInputs(
    time=data.index.tolist(),
    precipitation=data["P"].tolist(),
    evapotranspiration=data["E"].tolist(),
    catchment=catchment,
    run_period=ModelPeriod(start=start, end=end),
    # simulated run off and FDC will be exported as CSV files and figures
    destination=".",
    observed_runoff=data["Qmm"].tolist(),
    run_off_unit=RunOffUnit.NO_CONVERSION,
)

# Load the model and run it
model = GR6JModel(inputs)
results = model.run()

# Get the time and run-off vector as Pandas DataFrame use:
df = pd.DataFrame(zip(results.time, results.run_off), columns=["Time", "Run off"])
df.set_index("Time", inplace=True)
print(df)

# Get the exchange from routing store" for the only model and third time step use:
print(results.catchment_outputs[0][2].exchange_from_routing_store)
